src/Language/PureScript/Ide/CaseSplit.hs

```diff
@@ -131,15 +131,18 @@ parseType' s =
       throwError (GeneralError ("Parsing the splittype failed with:"
                                 ++ show err))
 
-parseTypeDeclaration' :: String -> (P.Ident, P.Type)
+parseTypeDeclaration' :: (MonadError PscIdeError m) => String -> m (P.Ident, P.Type)
 parseTypeDeclaration' s =
   let x = do
         ts <- P.lex "" s
         P.runTokenParser "" (P.parseDeclaration <* Parsec.eof) ts
   in
     case unwrapPositioned <$> x of
-      Right (P.TypeDeclaration i t) -> (i, t)
-      y -> error (show y)
+      Right (P.TypeDeclaration i t) -> pure (i, t)
+      Right _ -> throwError (GeneralError "Found a non-type-declaration")
+      Left err ->
+        throwError (GeneralError ("Parsing the typesignature failed with: "
+                                ++ show err))
```

Comment:
There is a function in `Error` somewhere for showing errors nicely. Does this get showed to the user?